### ISSUE TITLE
invalid plugin version 'cloudbees-bitbucket-branch-source:751.vda_24678a_f781', must follow pattern '^[0-9a-zA-Z+\\.-]+$'

### ISSUE THREAD

**Issue by mkyc** [jenkinsci/kubernetes-operator#711]
**Describe the bug**
Operator cannot use `cloudbees-bitbucket-branch-source:751.vda_24678a_f781` plugin version. 
**To Reproduce**

1. [deploy operator](https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/installing-the-operator/)
2. [deploy instance](https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/deploying-jenkins/)
3. add following section to configuration: 
   ```
    spec:
      master:
        basePlugins:
          - name: kubernetes
            version: "1.31.3"
          - name: workflow-job
            version: "2.42"
          - name: workflow-aggregator
            version: "2.6"
          - name: git
            version: "4.10.3"
          - name: job-dsl
            version: "1.78.3"
          - name: configuration-as-code
            version: "1.55.1"
          - name: kubernetes-credentials-provider
            version: "0.20"
          - name: aws-secrets-manager-credentials-provider
            version: "0.5.6"
          - name: branch-api
            version: "2.7.0"
          - name: cloudbees-bitbucket-branch-source
            version: "751.vda_24678a_f781"
   ```

**Additional information**

Kubernetes version: `v1.21.5-eks-bc4871b`
Jenkins Operator version: `v0.7.0`

```
2022-02-09T13:27:01.464Z	DEBUG	controller-jenkins	Reconciling Jenkins	{"cr": "master"}
2022-02-09T13:27:01.465Z	WARN	controller-jenkins	Validation of base configuration failed, please correct Jenkins CR.	{"cr": "master"}
github.com/go-logr/zapr.(*zapLogger).Info
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:126
github.com/jenkinsci/kubernetes-operator/controllers.(*JenkinsReconciler).reconcile
	/workspace/controllers/jenkins_controller.go:243
github.com/jenkinsci/kubernetes-operator/controllers.(*JenkinsReconciler).Reconcile
	/workspace/controllers/jenkins_controller.go:136
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:263
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:235
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:198
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.UntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99
2022-02-09T13:27:01.465Z	WARN	controller-jenkins	invalid plugin version 'cloudbees-bitbucket-branch-source:751.vda_24678a_f781', must follow pattern '^[0-9a-zA-Z+\\.-]+$'	{"cr": "master"}
github.com/go-logr/zapr.(*zapLogger).Info
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:126
github.com/jenkinsci/kubernetes-operator/controllers.(*JenkinsReconciler).reconcile
	/workspace/controllers/jenkins_controller.go:245
github.com/jenkinsci/kubernetes-operator/controllers.(*JenkinsReconciler).Reconcile
	/workspace/controllers/jenkins_controller.go:136
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:263
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:235
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:198
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.UntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99

```

**Anything else?**

I'm not sure if consider it operator issue or plugin versioning freestyle issue. I also reported it there: https://github.com/jenkinsci/bitbucket-branch-source-plugin/issues/556

**Comment by lemeurherve** [comment]
Duplicate of https://github.com/jenkinsci/kubernetes-operator/issues/704

**Comment by thecooldrop** [comment]
My question here to @Sig00rd is if it is even valid to have plugin-version validation logic in the operator? Operator does not care that version is properly formatted since:

- Improperly formatted should not even be publishable onto Jenkins plugin repositories
- If the version is not valid the startup will fail with an obvious error message

Current situation makes it harder to transparently see what went wrong. Maybe the operator should be changed to not validate versions anymore.

**Comment by yoshi314** [comment]
i am seeing the same thing with workflow-job : they went completely bonkers with versioning.

froim 2.42 to workflow-job:1174.vdcb_d054cf74a_

**Comment by thecooldrop** [comment]
> My question here to @Sig00rd is if it is even valid to have plugin-version validation logic in the operator? Operator does not care that version is properly formatted since:
> 
> * Improperly formatted should not even be publishable onto Jenkins plugin repositories
> * If the version is not valid the startup will fail with an obvious error message
> 
> Current situation makes it harder to transparently see what went wrong. Maybe the operator should be changed to not validate versions anymore.

Given the large number of up-votes here, including some from the maintainers, I will feel free to create a new issue with a request to remove the plugin-version validation logic and I would start working on it directly.

**Comment by thecooldrop** [comment]
This issue should be resolved once the PR #729 is merged. Maybe it can already be closed to reduce the issue backlog @Sig00rd ?

**Comment by brokenpip3** [comment]
this was resolved with https://github.com/jenkinsci/kubernetes-operator/pull/729, thanks @thecooldrop.
Closing
